plan/pattern.go

```diff
@@ -5,7 +5,7 @@ const AnyKind = "*** any procedure kind ***"
 // Pattern represents an operator tree pattern
 // It can match itself against a query plan
 type Pattern interface {
-	Root() ProcedureKind
+	Roots() []ProcedureKind
```

Comment:
This is an interesting way to do this. I like it.

An alternative way to implement a multiple kind pattern was to have this return `AnyKind` and then have match perform an actual check to see if it matches.

Reply:
I considered that approach as well as you mentioned it in a standup call the other day, but thought the more that happens in the pattern matcher, IE, the more that is declarative in nature, the better in the long run.